PythonVisualizations/Stack.py

```diff
@@ -0,0 +1,208 @@
+import time
+from tkinter import *
+try:
+    from drawable import *
+    from VisualizationApp import *
+except ModuleNotFoundError:
+    from .drawable import *
+    from .VisualizationApp import *
+
+CELL_SIZE = 40
+CELL_BORDER = 2
+CELL_BORDER_COLOR = 'black'
+STACK_X0 = 300
+STACK_Y0 =  400
+FONT_SIZE = 20
+VALUE_FONT = ('Helvetica', FONT_SIZE)
+VALUE_COLOR = 'black'
+
+
+
+# possible issues: 
+# when you enter an invalid number the operations grid moves slightly to the right (and then moves back when a valid number is entered)
+#  the cell_size was changed from 50 to 40 to show the entire stack vertically
+
+class Stack(VisualizationApp):
+    nextColor = 0
+
+    def __init__(self, size=10, title="Stack", **kwargs):
+        super().__init__(title=title, **kwargs)
+        self.size = size
+        self.title = title
+        self.list = []
+        self.buttons = self.makeButtons()
+
+  
+
+    def __str__(self):
+        return str(self.list)
+
+    # STACK FUNCTIONALITY
+
+   
+    def push(self, val):
+        self.cleanUp()
+      
+        # create new cell and cell value display objects
+        toPositions = (self.cellCoords(len(self.list)), 
+                       self.cellCenter(len(self.list)))
+
+        # Animate arrival of new value from operations panel area

```

Comment:
The comment here was probably copied from Array.py.  The animation for Stack has the new values sliding in from the top of the canvas.